### PR TITLE
arrow-cpp: add flight feature flag

### PR DESCRIPTION
**Pull request by cpcloud**
###### Motivation for this change

Add the option of building `arrow-cpp` with Flight enabled.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/arrow-cpp/default.nix [hunk 1]
```diff
@@ -1,8 +1,9 @@
 { stdenv, lib, fetchurl, fetchFromGitHub, fixDarwinDylibNames
-, autoconf, boost, brotli, cmake, flatbuffers, gflags, glog, gtest, lz4
-, perl, python3, rapidjson, re2, snappy, thrift, tzdata , utf8proc, which
-, zlib, zstd
+, autoconf, boost, brotli, c-ares, cmake, flatbuffers, gflags, glog, grpc
+, gtest, libnsl, lz4, openssl, perl, protobuf, python3, rapidjson, re2, snappy
+, thrift, tzdata, utf8proc, which, zlib, zstd
 , enableShared ? !stdenv.hostPlatform.isStatic
+, enableFlight ? true
```
Comment (veprbl) [inline pkgs/development/libraries/arrow-cpp/default.nix:6]:
```suggestion
, enableFlight ? !stdenv.isDarwin # libnsl is not supported on darwin
```

Comment (cpcloud) [reply pkgs/development/libraries/arrow-cpp/default.nix]: Done, thanks.

File: pkgs/development/libraries/arrow-cpp/default.nix [hunk 2]
```diff
@@ -126,27 +161,30 @@ in stdenv.mkDerivation rec {
...
   GTEST_FILTER =
-    if doInstallCheck then let
-      # Upstream Issue: https://issues.apache.org/jira/browse/ARROW-11398
-      filteredTests = lib.optionals stdenv.hostPlatform.isAarch64 [
-        "TestFilterKernelWithNumeric/3.CompareArrayAndFilterRandomNumeric"
-        "TestFilterKernelWithNumeric/7.CompareArrayAndFilterRandomNumeric"
-        "TestCompareKernel.PrimitiveRandomTests"
-      ];
-    in "-${builtins.concatStringsSep ":" filteredTests}" else null;
+    if doInstallCheck then
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/arrow-cpp/default.nix:164]: This should use some lib.optional variant.

Comment (cpcloud) [reply pkgs/development/libraries/arrow-cpp/default.nix]: Done.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 144609` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
21 packages built:
</summary>
<ul>
<li>arrow-cpp</li>
<li>python38Packages.datasets</li>
<li>python38Packages.google-cloud-bigquery</li>
<li>python38Packages.intake</li>
<li>python38Packages.intake-parquet</li>
<li>python38Packages.pyarrow</li>
<li>python38Packages.snowflake-connector-python</li>
<li>python38Packages.snowflake-sqlalchemy</li>
<li>python38Packages.spacy-transformers</li>
<li>python38Packages.tokenizers</li>
<li>python38Packages.transformers</li>
<li>python39Packages.datasets</li>
<li>python39Packages.google-cloud-bigquery</li>
<li>python39Packages.intake</li>
<li>python39Packages.intake-parquet</li>
<li>python39Packages.pyarrow</li>
<li>python39Packages.snowflake-connector-python</li>
<li>python39Packages.snowflake-sqlalchemy</li>
<li>python39Packages.spacy-transformers</li>
<li>python39Packages.tokenizers</li>
<li>python39Packages.transformers</li>
</ul>
</details>

**Pull request merged**